### PR TITLE
Add Spring Rest Template integration

### PR DESCRIPTION
**Pull request by choojoykin**
### Context
[//]: # (
)
Added support for spring rest template integration

#### Checklist
- [x] [Sign Allure CLA][cla]
- [x] Provide unit tests

[cla]: https://cla-assistant.io/accept/allure-framework/allure2

### CODE REVIEW

File: allure-spring-resttemplate/src/main/java/io/qameta/allure/resttemplate/AllureRestTemplate.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+public class AllureRestTemplate implements ClientHttpRequestInterceptor {
+
+    private ClientHttpResponse clientHttpResponse;
+    private String requestTemplatePath = "http-request.ftl";
+    private String responseTemplatePath = "http-response.ftl";
+
+    public AllureRestTemplate() {
+    }
+
+    public AllureRestTemplate(ClientHttpResponse clientHttpResponse) {
```
Comment (baev) [inline allure-spring-resttemplate/src/main/java/io/qameta/allure/resttemplate/AllureRestTemplate.java]: what is this constructor for

Comment (choojoykin) [reply allure-spring-resttemplate/src/main/java/io/qameta/allure/resttemplate/AllureRestTemplate.java]: ssory, its not required. deleted

File: allure-spring-resttemplate/src/main/java/io/qameta/allure/resttemplate/AllureRestTemplate.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * * Created by v.o.pavlov on 05.02.2019.
+ */
+public class AllureRestTemplate implements ClientHttpRequestInterceptor {
+
+    private ClientHttpResponse clientHttpResponse;
```
Comment (baev) [inline allure-spring-resttemplate/src/main/java/io/qameta/allure/resttemplate/AllureRestTemplate.java]: I guess this field unused as well

File: allure-spring-resttemplate/src/main/java/io/qameta/allure/resttemplate/AllureRestTemplate.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    private String requestTemplatePath = "http-request.ftl";
+    private String responseTemplatePath = "http-response.ftl";
+
+    public AllureRestTemplate() {
```
Comment (baev) [inline allure-spring-resttemplate/src/main/java/io/qameta/allure/resttemplate/AllureRestTemplate.java]: remove empty constructor

File: allure-spring-resttemplate/src/main/java/io/qameta/allure/resttemplate/AllureRestTemplate.java (hunk 2 above)
Comment (choojoykin) [reply allure-spring-resttemplate/src/main/java/io/qameta/allure/resttemplate/AllureRestTemplate.java]: fixed

File: allure-spring-resttemplate/src/main/java/io/qameta/allure/resttemplate/AllureRestTemplate.java (hunk 3 above)
Comment (choojoykin) [reply allure-spring-resttemplate/src/main/java/io/qameta/allure/resttemplate/AllureRestTemplate.java]: fixed

**Comment by baev** [comment]
@choojoykin 

> 3 Checkstyle rule violations were found in 1 files

> [Javadoc | JavadocStyle] io.qameta.allure.springweb.(AllureRestTemplate.java:22)
  First sentence should end with a period.
  http://checkstyle.sourceforge.net/config_javadoc.html#JavadocStyle

> [Sizes | LineLength] io.qameta.allure.springweb.(AllureRestTemplate.java:42)
  Line is longer than 120 characters (found 150).
  http://checkstyle.sourceforge.net/config_sizes.html#LineLength

> [Coding | FinalLocalVariable] io.qameta.allure.springweb.(AllureRestTemplate.java:56)
  Variable 'clientHttpResponse' should be declared final.
  http://checkstyle.sourceforge.net/config_coding.html#FinalLocalVariable

**Comment by baev** [comment]
@choojoykin 

> 3 PMD rule violations were found in 1 files

> [Code Style | MethodArgumentCouldBeFinal] io.qameta.allure.springweb.(AllureRestTemplate.java:42)
  Parameter 'body' is not assigned and could be declared final
  https://pmd.github.io/pmd-6.9.0/pmd_rules_java_codestyle.html#methodargumentcouldbefinal

> [Code Style | MethodArgumentCouldBeFinal] io.qameta.allure.springweb.(AllureRestTemplate.java:42)
  Parameter 'request' is not assigned and could be declared final
  https://pmd.github.io/pmd-6.9.0/pmd_rules_java_codestyle.html#methodargumentcouldbefinal

> [Code Style | MethodArgumentCouldBeFinal] io.qameta.allure.springweb.(AllureRestTemplate.java:43)
  Parameter 'execution' is not assigned and could be declared final
  https://pmd.github.io/pmd-6.9.0/pmd_rules_java_codestyle.html#methodargumentcouldbefinal

**Comment by choojoykin** [comment]
@baev build passed :)

**Comment by baev** [comment]
@choojoykin thanks!

**Pull request merged**